controllers/kuadrant_controller.go

```diff
@@ -386,14 +392,26 @@ func (r *KuadrantReconciler) createOnlyInKuadrantNSCb(ctx context.Context, kObj
 			return err
 		}
 
-		k8sObjKind := k8sObj.DeepCopyObject().GetObjectKind()
+		var newObj client.Object
+		newObj = k8sObj
+
+		switch obj := k8sObj.(type) {
+		case *appsv1.Deployment: // If it's a Deployment obj, it adds the required env vars
+			obj.Spec.Template.Spec.Containers[0].Env = []v1.EnvVar{
```

Comment:
What about appending instead if assigning?

```go
obj.Spec.Template.Spec.Containers[0].Env = append(obj.Spec.Template.Spec.Containers[0].Env, {Name: envLimitadorNamespace, Value: kObj.Namespace},{Name: envLimitadorName, Value: limitadorName})
```
Even if `obj.Spec.Template.Spec.Containers[0].Env` is `nil`, `append` operation works as expected.